pkg/operator/certrotation/metadata.go

```diff
@@ -1,36 +1,23 @@
 package certrotation
 
 import (
-	corev1 "k8s.io/api/core/v1"
+	"fmt"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func ensureOwnerRefAndTLSAnnotations(secret *corev1.Secret, owner *metav1.OwnerReference, additionalAnnotations AdditionalAnnotations) bool {
-	needsMetadataUpdate := false
+func ensureOwnerRefAndTLSAnnotations(meta *metav1.ObjectMeta, owner *metav1.OwnerReference, additionalAnnotations AdditionalAnnotations) string {
```

Comment:
this could return []string

Reply:
Good idea, this simplifies quite a few things (although its rare that an update changes both owners and annotations during normal update)